File: practica1/Ejercicio1/src/ejercicio1/FactoriaCarretera.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio1;

/**
 *
 * @author carmenxufdz
 */
public class FactoriaCarretera implements FactoriaCarreraYBicicleta {

    @Override
    public Carrera crearCarrera() {
        return new CarreraCarretera();
    }

    @Override
    public Bicicleta crearBicicleta(int id) {
        return new BicicletaCarretera(id);
    }
    
}

File: practica1/Ejercicio2/FactoriaCarretera.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from carreraCarretera import CarreraCarretera
from BicicletaCarretera import BicicletaCarretera

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crearCarrera(bicicletas):
        return CarreraCarretera(bicicletas, porcent_retirada=0)

    def crearBicicleta(id):
        return BicicletaCarretera(id)